lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -130,8 +130,8 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
     metadataStoreParams.dataStoreName = location;
 
     // NOTE: remove the following when we will support putting a
-    // versioned object to a location-constraint of type AWS or Azure.
-    if (locationType && externalBackends[locationType]) {
+    // versioned object to a location-constraint of type Azure.
+    if (locationType === 'azure') {
```

Comment:
I know it was already wrong before this PR, but this code must ask the data backend directly to know if it supports versioning. this should be considered a different issue

Reply:
instead of saying azure, let's have some sort of constant object for backends with unsupported versioning with azure set to true.